From 2299a60a896a1ca81347e6ef08d9cc6f9dc0fa9b Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Mon, 5 Dec 2016 11:33:01 +0200
Subject: [PATCH 0631/1345] fix: dts: a8040: Add missing SPI config in DB-A
 DTS

commit  223be8bf6a68cc65f22fd1be4079f34e9ac9f70f from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Add SPI flash on CP1 configuration to armada-8040-db-A.dts
Add spi1 pins configuration to armada-8040.dtsi
Fixes Jira bug SYSTEMSW-3080

Change-Id: I06aadcbc45296a661b0c43c53c61fa744390b92a
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/34344
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-db-A.dts |   24 ++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
index b31f19ff..8ed6aa4 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
@@ -188,6 +188,30 @@
 			comphy {
 				status = "okay";
 			};
+			spi1: spi@700680 {
+				status = "okay";
+				pinctrl-names = "default";
+				pinctrl-0 = <&cps_spi1_pins>;
+				spi-flash@0 {
+					#address-cells = <0x1>;
+					#size-cells = <0x1>;
+					compatible = "jedec,spi-nor";
+					reg = <0x0>;
+					spi-max-frequency = <20000000>;
+					partition@0 {
+						label = "boot";
+						reg = <0x0 0x200000>;
+					};
+					partition@200000 {
+						label = "Filesystem";
+						reg = <0x200000 0xd00000>;
+					};
+					partition@d00000 {
+						label = "boot_2nd";
+						reg = <0xf00000 0x100000>;
+					};
+				};
+			};
 		};
 		pcie@0x600000 {
 			status = "okay";
-- 
1.7.9.5

